network_model/network modeling/comparing_towers.py

```diff
@@ -0,0 +1,191 @@
+        dist = factor
+        if signal_loss < 0:
+            add = True
+        cont = True
+        power = 0
+        while cont:
+            if factor <= Decimal(factor_change) / Decimal(math.pow(10, power)):
+                power += 1
+            else:
+                cont = False
+        increase_power += 1 if add == prev_two_adds[1] and add != prev_two_adds[0] else 0
+        amount_to_change = Decimal(factor_change) / Decimal(math.pow(10, power + increase_power))
+        factor += amount_to_change if add is True else amount_to_change * -1
+        if change_add_index == True:
+            prev_two_adds[0] = add
+        else:
+            prev_two_adds[1] = add
+    return dist.quantize(Decimal('1.000000000000000000000000')), signal_loss
+
+
+def convert_miles_to_km(distance):
+    return distance * 1.609344
+
+
+def open_csv(name):
+    try:
+        with open(name) as f:
+            reader = csv.reader(f)
+            data = list(reader)
+        data = data[1:] #remove column names
+        return data
+    except:
+        print("invalid file name entered into open_csv")
+        return 0
+
+
+def clean_tigo_data(data):
+    #remove incomplete data
+    return filter(lambda (a, b, c): b != "TBD" and b != "NA" and c != "TBD" and c != "NA", data)
+
+
+def clean_open_cell_id_data(data):
+    #remove non-tigo data
+    return filter(lambda (a, b, c, d, e, f, g): c == '2', data)
+
+
+def get_columns(data, cols):
+    #cols should be a list of the indices of the columns wanted from data
+    try:
+        mod_data = list()
+        for x in data:
+            holder = list()
+            for col in cols:
+                holder.append(x[col])
+            mod_data.append(holder)
+        return mod_data
+    except:
+        print("invalid parameters entered into get_columns")
+        return 0
+
+
+def distance_between_coords(lat1, lon1, lat2, lon2):
+    lat_mid = (lat1 + lat2) / 2
+    lon_mid = (lon1 + lon2) / 2
+    meters_per_deg_lat = 111132.954 - 559.822 * math.cos(2 * lat_mid) + 1.175 * math.cos(4 * lat_mid)
+    meters_per_deg_lon = (3.14159265359 / 180) * 6367449 * math.cos(lat_mid)
+    delta_lat = math.fabs(lat1 - lat2)
+    delta_lon = math.fabs(lon1 - lon2)
+    return math.sqrt(math.pow(delta_lat * meters_per_deg_lat, 2) + math.pow(delta_lon * meters_per_deg_lon, 2))
+
+
+def long_cid_to_short_cid(long_cid):
+    #modulus reduction to get last 16 bits from long_cid to make the short_cid
+    return long_cid % (2**16)
+
+
+def num_to_bin(num):
+    #return binary representation of number
+    return bin(int(num))
+
+
+def num_to_bin_str(num):
+    #return the binary representation of a number (split up by groups of four bits for readability)
+    binary_str = str(bin(int(num)))[2:]
+    mod_str = ""
+    for x in range(0, len(binary_str)):
+        if (x + 1) % 5 == 0:
+            mod_str += " "
+        else:
+            mod_str += binary_str[x]
+    return mod_str
+
+
+def put_cell_id_together(cell):
```

Comment:
rename to indicate that this converts from `open_cell_id` to `tigo_id` [CHANGE REQUESTED]